cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfBotPawn.cpp

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfBotPawn.h"
+
+#include <Camera/CameraComponent.h>
+#include <Components/InputComponent.h>
+
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Unreal.h"
+#include "UrdfRobotComponent.h"
+#include "UrdfParser.h"
+
+AUrdfBotPawn::AUrdfBotPawn(const FObjectInitializer& object_initializer): APawn(object_initializer)
+{
+
+    // setup robot
+    std::string urdf_file = Unreal::toString(FPaths::Combine(
+        Unreal::toFString(Config::get<std::string>("URDFBOT.URDFBOT_PAWN.URDF_DIR")),
+        Unreal::toFString(Config::get<std::string>("URDFBOT.URDFBOT_PAWN.URDF_FILE"))));
+    UrdfRobotDesc robot_desc = UrdfParser::parse(urdf_file);
+
+    robot_component_ = CreateDefaultSubobject<UUrdfRobotComponent>(TEXT("RobotComponent"));
+    robot_component_->initialize(&(robot_desc));
```

Comment:
```
robot_component_ = CreateDefaultSubobject<UUrdfRobotComponent>(Unreal::toFName("AUrdfBotPawn::robot_component_::" + robot_desc.name_));
robot_component_->initializeComponent(&robot_desc);
robot_component_->createChildComponents(&robot_desc);

RootComponent = robot_component_;
```

Reply:
Do we want to use `robot_component` or `root_link_component` as RootComponent? The `robot_component` is a virtual component and its transform do not follow physical simulation. It will affect `camera_component` attaching target, and the result from `AUrdfBotPawn::GetActorTransform`.